source/layouts/repo_layout.html.erb

```diff
@@ -1,26 +1,37 @@
 <% content_for :sidebar do %>
-  <ul>
-    <li>
-      <a href="/apps/by-team.html">
-        <span>Repository ownership by team</span>
-      </a>
-    </li>
+  <%
+    items = [
+      sanitize("<a href=\"/repos/#{repo.repo_name}.html\" rel=\"noopener\" class=\"govuk-link\">#{repo.repo_name}</a>"),
+    ]
 
-    <% Repos.public.group_by(&:type).each do |name, apps| %>
-      <li>
-        <a href="/apps.html">
-          <span><%= name %></span>
-        </a>
-      </li>
-      <li>
-        <ul class='subnav'>
-        <% apps.each do |app| %>
-          <li><%= sidebar_link app.repo_name, "/repos/#{app.repo_name}.html" %></li>
-        <% end %>
-        </ul>
-      </li>
-    <% end %>
-  </ul>
+    unless repo.private_repo?
+      items << sanitize("<a href=\"/repos/#{repo.repo_name}.html#readme\" rel=\"noopener\" class=\"govuk-link\">README</a>")
+
+      add_imported_docs = -> (docs_to_import) do
+        docs_subdirs = docs_to_import.map { |doc| doc[:path].split("/")[3..(doc[:path].split("/").count - 2)].join("/") }.uniq.reject(&:empty?)
+        docs_at_root_level = docs_to_import.reject do |doc|
+          docs_subdirs.any? { |subdir| doc[:path].include?(subdir) }
+        end
+        docs_at_root_level.each do |doc|
+          items << sanitize("<a href='#{doc[:path]}' rel='noopener' class='govuk-link'>#{doc[:title]}</a>")
+        end
+        docs_subdirs.each do |subdir|
+          items << sanitize("<strong>#{subdir}</strong>")
+          subdir_items = []
+          docs_to_import.select { |doc| doc[:path].include?(subdir) }.each do |doc|
+            subdir_items << sanitize("<a href='#{doc[:path]}' rel='noopener' class='govuk-link'>#{doc[:title]}</a>")
+          end
+          items << GovukPublishingComponents.render("govuk_publishing_components/components/list", { visible_counters: true, items: subdir_items })
+        end
+      end
+
+      imported_docs = GitHubRepoFetcher.instance.docs(repo.repo_name)
+      unless imported_docs.nil? || imported_docs.empty?
+        add_imported_docs.call(imported_docs)
+      end
+    end
```

Comment:
☝️  this all looks painful but that's mostly down to the lack of syntax highlighting! This is basically a lift and shift of what was already in `source/partials/repo/_imported_docs.html.erb` (now deleted), converted to pure Ruby (rather than ERB) as we're now using a component to render the data.